### PR TITLE
Add unit test to hive query logging

### PR DESCRIPTION
**Pull request by dcfocus**
This pull request added unit test to hive query logging. 
1. Refactor code to be testable: decoupled locally created variables by adding parameters at interface, and changed **twitterscriber** to be **Protected** from **Private**. 
2. Remove unused fields in QueryStats: currentTimeStamp and mapReduceStatsDesc, and their accessor methods.
3. Clean up confusing variable names: renamed thriftCounterInfo to thriftStageInfo. 
4. Add unit test.

### CODE REVIEW

File: ql/src/test/org/apache/hadoop/hive/ql/history/TestHiveScribeImpl.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    assertEquals(taskEnd, hiveScriber.getQueryStatsMap().get(queryId).getTaskProgress().get(2).getProgress());
+
+    // Progress Plan
+    hiveScriber.logPlanProgress(plan, planTime);
+
+    // End query
+    hiveScriber.endQuery(queryId, sessionState, queryEndTime);
+
+    // Capture log event at scribe()
+    ArgumentCaptor<HiveQueryCompletionEvent> arg = ArgumentCaptor.forClass(HiveQueryCompletionEvent.class);
```
Comment (Yaliang) [inline ql/src/test/org/apache/hadoop/hive/ql/history/TestHiveScribeImpl.java]: I would prefer capture a string here and then decode string as a `HiveQueryCompletionEvent` and finally compare the fields.

Comment (dcfocus) [reply ql/src/test/org/apache/hadoop/hive/ql/history/TestHiveScribeImpl.java]: Updated Mockito capture to return a String, and then decode it to a thrift object and compare.

**Review by Yaliang — APPROVED** [review]
LGTM, Thanks

**Comment by Yaliang** [comment]
Merged!

**Pull request merged**